src/huggingface_hub/constants.py

```diff
@@ -65,7 +65,7 @@ def _as_int(value: Optional[str]) -> Optional[int]:
 _HF_DEFAULT_STAGING_ENDPOINT = "https://hub-ci.huggingface.co"
 ENDPOINT = os.getenv("HF_ENDPOINT") or (_HF_DEFAULT_STAGING_ENDPOINT if _staging_mode else _HF_DEFAULT_ENDPOINT)
```

Comment:
```suggestion
ENDPOINT = os.getenv("HF_ENDPOINT", "").rstrip("/") or (
    _HF_DEFAULT_STAGING_ENDPOINT if _staging_mode else _HF_DEFAULT_ENDPOINT
)
```
I'd rather normalize `ENDPOINT` here as it is used throughout the codebase.